### PR TITLE
fix: Do not search for simulators in wrong region

### PR DESCRIPTION
**Pull request by speller26**
Do not call search_devices at all if requesting simulators only and in a different region.

*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/aws/aws_device.py [hunk 1]
```diff
@@ -399,31 +401,34 @@ def get_devices(
...
-            region_device_arns = [
-                result["deviceArn"]
-                for result in session_for_region.search_devices(
-                    arns=arns,
-                    names=names,
-                    types=region_device_types,
-                    statuses=statuses,
-                    provider_names=provider_names,
+            session_region = aws_session.boto_session.region_name
+            # If the region is not the same as the current region's
```
Comment (licedric) [inline src/braket/aws/aws_device.py:405]:
```suggestion
            # If the region is not the same as the current region
```

**Review by licedric — APPROVED** [review]

File: src/braket/aws/aws_device.py [hunk 2]
```diff
@@ -399,31 +401,34 @@ def get_devices(
...
+            # If the region is not the same as the current session's region
+            # and only simulators are requested, then search_devices will not
+            # be called at all because simulators are only retrieved for the
+            # current region
+            if region == session_region or types != {AwsDeviceType.SIMULATOR}:
+                session_for_region = AwsDevice._copy_aws_session(aws_session, [region])
+                # Simulators are only instantiated in the same region as the AWS session
+                region_device_types = sorted(
+                    types if region == session_region else types - {AwsDeviceType.SIMULATOR}
+                )
```
Comment (kshitijc) [inline src/braket/aws/aws_device.py:414]: Is this now redundant?

**Review by kshitijc — COMMENTED** [review]
Would there be value in having integration tests for get_devices?

File: src/braket/aws/aws_device.py (hunk 2 above)
Comment (speller26) [reply src/braket/aws/aws_device.py:414]: Nope, because it's still possible to ask for both QPUs and simulators, in which case on QPUs will be searched

**Pull request merged**